src/app/game.ts

```diff
@@ -321,7 +321,7 @@ export class Game {
   }
 
   private handlewWallWasBuilt(wall: Wall) {
-    this.villageDefence += wall.defense
+    this.villageDefence += wall.defensePerLevel[wall.getLevel()]
```

Comment:
Here better would be to create a function in wall class, e.g. getDefence and there return just this.defensePerLevel[this.level] and variable defencePerLevel can be then private. (Change everywhere defese to defence)